encoding/string/encoding.go

```diff
@@ -0,0 +1,46 @@
+package string
+
+import "io"
+
+// Encoder encodes string values.
+type Encoder interface {
+	// Encode a collection of strings.
+	// Callers should explicitly call `Reset` before
+	// subsequent call to `Encode`.
+	Encode(Iterator) error
```

Comment:
Same comments as those in the `int/encoding.go`

Reply:
Ping, should add var name here.

Reply:
👍